ui/app/AppLayouts/Profile/panels/ContactsListPanel.qml

```diff
@@ -36,79 +37,62 @@ Item {
-                if (panelUsage === Constants.contactsPanelUsage.verifiedMutualContacts) {
-                    if(item.model.isVerified)
-                        visible.push(item);
-                }
-                else if(panelUsage === Constants.contactsPanelUsage.mutualContacts) {
-                    if(!item.model.isVerified)
-                        visible.push(item);
-                }
-                else {
-                    visible.push(item);
-                }
-            }
-
-            for (i = 0; i < visible.length; ++i) {
-                item = visible[i];
-                item.inVisible = true;
-                if (item.visibleIndex !== i) {
-                    visibleItems.move(item.visibleIndex, i, 1);
-                }
-            }
-        }
-
-        model: contactListRoot.contactsModel
-
-        groups: [DelegateModelGroup {
-                id: visibleItems
-                name: "visible"
-                includeByDefault: false
-            }]
-
-        filterOnGroup: "visible"
-        items.onChanged: update()
-        delegate: contactPanelComponent
-    }
-
     StatusListView {
         id: contactsList
         anchors.top: title.bottom
         anchors.left: parent.left
         anchors.right: parent.right
         anchors.bottom: parent.bottom
         interactive: false
-        ScrollBar.vertical.policy: contactListRoot.scrollbarOn ? ScrollBar.AlwaysOn : ScrollBar.AsNeeded
-        model: delegateModel
-    }
+        ScrollBar.vertical.policy: root.scrollbarOn ? ScrollBar.AlwaysOn : ScrollBar.AsNeeded
+        model: SortFilterProxyModel {
+            id: filteredModel
+
+            sourceModel: root.contactsModel
+
+            function panelUsagePredicate(isVerified) {
+                if (panelUsage === Constants.contactsPanelUsage.verifiedMutualContacts) return isVerified
+                if (panelUsage === Constants.contactsPanelUsage.mutualContacts) return !isVerified
+                return true
+            }
+
+            function searchPredicate(name, pubkey) {
+                if (root.searchString === "") return true
+
+                let lowerCaseSearchString = root.searchString.toLowerCase()
+                let compressedPubkey = Utils.getCompressedPk(pubkey)
 
-    Component {
-        id: contactPanelComponent
+                return name.toLowerCase().includes(lowerCaseSearchString) ||
+                       pubkey.toLowerCase().includes(lowerCaseSearchString) ||
+                       compressedPubkey.toLowerCase().includes(lowerCaseSearchString)
+            }
+
+            filters: [
+                ExpressionFilter { expression: filteredModel.panelUsagePredicate(model.isVerified) },
+                ExpressionFilter {
+                    expression: {
+                        root.searchString // ensure expression is reevaluated when searchString changes
+                        filteredModel.searchPredicate(model.displayName, model.pubKey)
```

Comment:
shouldn't it be preceded by a `return`?

Reply:
it should :sweat_smile: dunno how it worked before